core-web/libs/data-access/src/lib/ema-app-configuration/ema-app-configuration.service.ts

```diff
@@ -28,68 +28,77 @@ export class EmaAppConfigurationService {
-            takeWhile((isEnterprise: boolean) => isEnterprise), // stop if license is not enterprise
-            switchMap(() => this.getCurrentSiteIdentifier()),
-            switchMap((currentSiteId: string) =>
-                this.getEmaAppConfiguration(currentSiteId).pipe(
-                    map((appConfiguration) => ({ currentSiteId, appConfiguration }))
-                )
-            ),
-            map(({ currentSiteId, appConfiguration }) =>
-                this.getConfigurationForCurrentSite(appConfiguration, currentSiteId)
-            ),
-            takeWhile((site: DotAppsSite | null) => !!site), // stop if site is undefined or null
-            map(getSecretByUrlMatch(url)),
-            defaultIfEmpty<SecretValue | null>(null)
-        );
+    siteService = inject(SiteService);
+
+    /**
+     * Get the EMA app configuration for the current site.
+     *
+     * @param {string} url
+     * @return {*}  {(Observable<EmaAppSecretValue | null>)}
+     * @memberof EmaAppConfigurationService
+     */
+    get(url: string): Observable<EmaAppSecretValue | null> {
+        return this.licenseService
+            .isEnterprise()
+            .pipe(catchError(() => EMPTY))
+            .pipe(
+                switchMap(() => this.getCurrentSiteIdentifier()),
+                switchMap((currentSiteId: string) =>
+                    this.getEmaAppConfiguration(currentSiteId).pipe(
+                        map(getConfigurationForCurrentSite(currentSiteId))
+                    )
+                ),
+                map(getSecretByUrlMatch(url)),
+                catchError(() => EMPTY),
+                defaultIfEmpty<EmaAppSecretValue | null>(null)
+            );
     }
 
     private getCurrentSiteIdentifier(): Observable<string> {
-        return this.http
-            .get<{ entity: Site }>('/api/v1/site/currentSite')
-            .pipe(pluck('entity', 'identifier'));
+        return this.siteService.getCurrentSite().pipe(pluck('identifier'));
     }
 
-    private getEmaAppConfiguration(id: string): Observable<DotApps> {
+    private getEmaAppConfiguration(id: string): Observable<DotApp> {
         return this.http
-            .get<{ entity: DotApps }>(`/api/v1/apps/dotema-config-v2/${id}`)
+            .get<{ entity: DotApp }>(`/api/v1/apps/dotema-config-v2/${id}`)
             .pipe(pluck('entity'));
     }
+}
 
-    private getConfigurationForCurrentSite(
-        appConfiguration: DotApps,
-        siteId: string
-    ): DotAppsSite | null {
-        return (
-            appConfiguration?.sites?.find((site) => site.id === siteId && site.configured) || null
-        );
-    }
+function getConfigurationForCurrentSite(currentSiteId: string): (app: DotApp) => DotAppsSite {
+    return (app) => {
+        const site =
+            app?.sites?.find((site) => site.id === currentSiteId && site.configured) || null;
+
+        if (!site) {
+            throw new Error('No site configuration found');
```

Comment:
does this need translation?

Reply:
We don't need to translate errors unless we show them in the UI and even there if the error comes from the backend it have no translation.